### ISSUE TITLE
deploy Yosys in nodejs

### ISSUE THREAD

**Issue by Bestduan** [YosysHQ/yosys#2761]
Hello, these days I plan to deploy Yosys in VScode, but I have encountered some problems. The first is whether Yosys.js can be started and used in native JavaScript?
The following problems are encountered during debugging：
![image](https://user-images.githubusercontent.com/38045864/118686982-1cc02a80-b837-11eb-8413-bf6b68200dc2.png)
How can I export the functions in Yosys.js and use them in native JavaScript?
I look forward to your clarification.

**Comment by racerxdl** [comment]
Since that's a web library, try just requiring it without assigning to anything.

```javascript
require("./yosysjs");
console.log(YosysJS);
```

In browser-js, things are declared under global.XXXX or window.XXXX. In case of nodejs it probably goes under global.XXXX which will maps directly as XXXXX.

You can also edit `yosys.js` file and add to the end:

```javascript
exports.YosysJS = YosysJS;
```

Which should make it work like you tried.

**Comment by Bestduan** [comment]
First of all, Yosys.js does not contain YosysJS objects. In addition, after that, I tried to use the methods exposed in Yosys.js directly and successfully read the Verilog file and completed some designs. However, I encountered problems in how to export the design.I can't export the designed file.The specific code is as follows:
`
var fs = require("fs");
var yosys   = require("../yosys-no-wasm-0.9/yosys");
console.log(yosys);

// yosys.FS.writeFile("demo.v", fs.readFileSync("./demo.v", "utf-8"), { encoding: 'utf8' });
let demo_text = fs.readFileSync("D:/project/Code/Javascript/yosysjs/demo.v", "utf-8");
yosys.FS_createPath("/","verilog",true,true)
yosys.FS_createDataFile("/verilog","demo.v", demo_text,true,true,false);
yosys.ccall('run', '', ['string'], ["read_verilog /verilog/demo.v"]);
yosys.ccall('run', '', ['string'], ["synth_anlogic -top demo"]);
yosys.ccall('run', '', ['string'], ["write_verilog /verilog/full.v"]);
`
The first problems encountered are as follows：
![image](https://user-images.githubusercontent.com/38045864/123556799-f390a380-d7bf-11eb-8d0d-c1f55de09450.png)
How can I fix it?
Second, how do I get the contents of the Full.v file if I do it correctly?
Being able to run Yosys on NodeJS is really important for my future work. Looking forward to your reply.
